libs/wire-api-federation/test/Test/Wire/API/Federation/ClientSpec.hs

```diff
@@ -47,128 +45,141 @@ import Wire.API.Federation.GRPC.Types (Component (Brig), FederatedRequest (Feder
 import Wire.API.User (UserProfile)
 
 spec :: Spec
-spec = xdescribe "Federator.Client" $ do
-  it "should make correct calls to the federator and parse success response correctly" $ do
-    handle <- generate arbitrary
-    expectedResponse :: Maybe UserProfile <- generate arbitrary
+spec = do
+  stateRef <- runIO . newIORef $ MockState [] (error "No mock response provided") (error "server not started") (error "No port selected yet")
+  beforeAll (startMockFederator stateRef) $
+    afterAll_ (stopMockFederator stateRef) $
+      before_ (flushState stateRef) $
+        describe "Federator.Client" $ do
```

Comment:
Minor nitpick: Would there be a way to do these nice things in a way that doesn't lead to so much nesting/indentation for all the individual tests?

Reply:
Hmm, maybe using `.` to compose these instead of `$` would make ormolu format them nicely, something like this:

```haskell
  beforeAll (startMockFederator stateRef)
    . afterAll_ (stopMockFederator stateRef)
    . before_ (flushState stateRef)
    $ describe "Federator.Client" $ do
      it "should make correct calls to the federator and parse success response correctly" $ do
```

I don't think there is a way to just have these hooks setup separate from the test blocks (and not wrapping them) like JUnit, RSpec etc.

Reply:
Done here:  bb2b627